ambari-server/src/main/java/org/apache/ambari/server/mpack/MpackManager.java

```diff
@@ -564,10 +544,7 @@ protected void populateStackDB(Mpack mpack) throws IOException {
       stackEntity.setMpackId(mpack.getMpackId());
       stackDAO.create(stackEntity);
     } else {
-      LOG.info("Updating stack {}-{} to the database", stackName, stackVersion);
-
-      stackEntity.setMpackId(mpack.getMpackId());
-      stackDAO.merge(stackEntity);
+      LOG.error("Stack {}-{} already exists in the database", stackName, stackVersion);
```

Comment:
We should through an exception here. Infact we should add a precheck to ensure that corresponding stack doesn't exist in the DB BEFORE creating an mpack DB entry.

Reply:
Hi Jayush, since the flow during registration is always populateMpackDB and then populateStackDB, there is a check in populateMpackDB to see if the entry already exists in table mpacks, only then it is added. While deleting an mpack also first stackEntry is removed and then the MpackEntry. Will that not take care of the scenario? Kindly advise. I will modify code to throw an exception instead of Logging an error.